packages/sales-api-service/src/services/__tests__/concession.service.spec.js

```diff
@@ -0,0 +1,283 @@
+import { CONCESSION, CONCESSION_PROOF } from '../constants.js'
+import { getReferenceDataForEntity, getReferenceDataForEntityAndId } from '../reference-data.service.js'
+
+jest.mock('@defra-fish/connectors-lib')
+jest.mock('../reference-data.service.js')
+
+getReferenceDataForEntity.mockImplementation(() => [
+  {
+    id: 'aaa-111-bbb-222',
+    name: CONCESSION.JUNIOR
+  },
+  {
+    id: 'ccc-333-ddd-444',
+    name: CONCESSION.SENIOR
+  },
+  {
+    id: 'eee-555-fff-666',
+    name: CONCESSION.DISABLED
+  }
+])
+
+getReferenceDataForEntityAndId.mockImplementation((_e, id) => {
+  if (id === 'aaa-111-bbb-222') {
+    return {
+      id,
+      name: CONCESSION.JUNIOR
+    }
+  }
+  if (id === 'ccc-333-ddd-444') {
+    return {
+      id,
+      name: CONCESSION.SENIOR
+    }
+  }
+  if (id === 'eee-555-fff-666') {
+    return {
+      id,
+      name: CONCESSION.DISABLED
+    }
+  }
+  throw Error('Unrecognised concession id')
+})
+
+const getMockPermission = () => ({})
+
+const getJuniorConcession = () => ({
+  id: 'aaa-111-bbb-222',
+  name: CONCESSION.JUNIOR,
+  proof: {
+    type: CONCESSION_PROOF.none
+  }
+})
+
+const getSeniorConcession = () => ({
+  id: 'ccc-333-ddd-444',
+  name: CONCESSION.SENIOR,
+  proof: {
+    type: CONCESSION_PROOF.none
+  }
+})
+
+const getDisabledBlueBadgeConcession = () => ({
+  id: 'eee-555-fff-666',
+  name: CONCESSION.DISABLED,
+  proof: {
+    type: CONCESSION_PROOF.blueBadge,
+    referenceNumber: 'blue-badge-123'
+  }
+})
+
+const getDisabledNiConcession = () => ({
+  id: 'eee-555-fff-666',
+  name: CONCESSION.DISABLED,
+  proof: {
+    type: CONCESSION_PROOF.NI,
+    referenceNumber: 'national-insurance-456'
+  }
+})
+
+describe('preparePermissionDataForRenewal', () => {
+  describe('Disabled', () => {
+    describe('addDisabled', () => {
+      it('add blue badge', async () => {
+        const permission = getMockPermission()
+        await concessionService.addDisabled(permission, CONCESSION_PROOF.blueBadge, 'bb-111')
+        expect(permission.concessions).toContainEqual(
+          expect.objectContaining({
+            id: 'eee-555-fff-666',
+            name: CONCESSION.DISABLED,
+            proof: {
+              type: CONCESSION_PROOF.blueBadge,
+              referenceNumber: 'bb-111'
+            }
+          })
+        )
+      })
+
+      it('add NI', async () => {
+        const permission = getMockPermission()
+        await concessionService.addDisabled(permission, CONCESSION_PROOF.NI, 'national-insurance-456')
+        expect(permission.concessions).toContainEqual(expect.objectContaining(getDisabledNiConcession()))
+      })
+
+      it('add blue badge to senior licence', async () => {
+        const permission = getMockPermission()
+        permission.concessions = [getSeniorConcession()]
+        await concessionService.addDisabled(permission, CONCESSION_PROOF.blueBadge, 'blue-badge-123')
+        expect(permission.concessions).toContainEqual(getDisabledBlueBadgeConcession())
```

Comment:
Can we also test that the senior concession is still there, like we do when adding a senior concession to a disabled concession (so the other way round)?

Same goes for test on line 116.